src/Query/TaskGroupingTree.ts

```diff
@@ -40,38 +32,22 @@ class TaskGroupingTreeNode extends GroupingTreeNode<Task> {}
  * TaskGroupingTree does the initial grouping together of tasks,
  * in arbitrary order. Caller is responsible for sorting groups in to desired order.
  *
- * It is essentially a thin wrapper around Map - see {@link TaskGroupingTreeStorage}.
- *
- * It is an implementation detail of the task-grouping code, and does not need to
- * be understood in order to group tasks.
- *
- * Ideally, this code would be simplified and moved in to TaskGroups.
  */
 export class TaskGroupingTree {
-    public groups = new TaskGroupingTreeStorage();
+    private root: TaskGroupingTreeNode;
 
     /**
-     * Group a list of tasks, according to one or more task properties
+     * The tree is built layer by layer, starting from the root.
+     * At every level, we iterate on the nodes of that level to generate
+     * the next one using the next grouping. The tree tree groups
+     * the passed {@link tasks} by the passed {@link groupers}.
```

Comment:
> It's not obvious to me that lines 40-43 are useful for the caller to know - but maybe I misunderstood something.

I rollback some comments, let me know if this is better. I agree I went to aggressive to them =)

> What was the reason for inlining buildGroupingTree()?

Because now the constructor would be just one line: `root = buildGroupingTree()`. What's the point of having that? I thought it would make sense to inline since there is nothing else.

Reply:
> What's the point of having that? I thought it would make sense to inline since there is nothing else.

The point of having the function is to give the behaviour a readable name - rather than having to read several lines of comments to figure out what is going on.

I don't have time to wait for another cycle, so I'll make the change and push it before merging...

Reply:
> The point of having the function is to give the behaviour a readable name - rather than having to read several lines of comments to figure out what is going on.

I agree and I thought that `TaskGroupingTree.constructor()` is explicit enough and doesn't need a `buildTaskGroupingTree` inside.

In any case I'm glad we found some middle ground for this.